src/main/scala/mesosphere/marathon/core/launchqueue/impl/RateLimiterActor.scala

```diff
@@ -0,0 +1,145 @@
+import org.apache.mesos.Protos.TaskID
+import rx.lang.scala.Subscription
+import scala.concurrent.duration._
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+private[launchqueue] object RateLimiterActor {
+  def props(
+    rateLimiter: RateLimiter,
+    taskTracker: TaskTracker,
+    appRepository: AppRepository,
+    updateReceiver: ActorRef,
+    taskStatusObservables: TaskStatusObservables): Props =
+    Props(new RateLimiterActor(
+      rateLimiter, taskTracker, appRepository, updateReceiver, taskStatusObservables
+    ))
+
+  case class DelayUpdate(app: AppDefinition, delayUntil: Timestamp)
+
+  case class ResetDelay(app: AppDefinition)
+  case object ResetDelayResponse
+
+  case class GetDelay(appDefinition: AppDefinition)
+  private[impl] case class AddDelay(app: AppDefinition)
+  private[impl] case class DecreaseDelay(app: AppDefinition)
+
+  private case object CleanupOverdueDelays
+}
+
+private class RateLimiterActor private (
+    rateLimiter: RateLimiter,
+    taskTracker: TaskTracker,
+    appRepository: AppRepository,
+    updateReceiver: ActorRef,
+    taskStatusObservables: TaskStatusObservables) extends Actor with ActorLogging {
+  var taskStatusSubscription: Subscription = _
+  var cleanup: Cancellable = _
+
+  override def preStart(): Unit = {
+    taskStatusSubscription = taskStatusObservables.forAll.subscribe(self ! _)
+    import context.dispatcher
+    cleanup = context.system.scheduler.schedule(10.seconds, 10.seconds, self, CleanupOverdueDelays)
+    log.info("started RateLimiterActor")
+  }
+
+  override def postStop(): Unit = {
+    taskStatusSubscription.unsubscribe()
+    cleanup.cancel()
+  }
+
+  override def receive: Receive = LoggingReceive {
+    Seq[Receive](
+      receiveCleanup,
+      receiveDelayOps,
+      receiveTaskStatusUpdate
+    ).reduceLeft(_.orElse[Any, Unit](_))
+  }
+
+  private[this] def receiveCleanup: Receive = {
+    case CleanupOverdueDelays =>
+      // If an app gets removed or updated, the delay should be reset.
+      // Still, we can remove overdue delays before that and also make leaks less likely
+      // by calling this periodically.
+      rateLimiter.cleanUpOverdueDelays()
+  }
+
+  private[this] def receiveDelayOps: Receive = {
+    case GetDelay(app) =>
+      sender() ! DelayUpdate(app, rateLimiter.getDelay(app))
+
+    case AddDelay(app) =>
+      rateLimiter.addDelay(app)
+      updateReceiver ! DelayUpdate(app, rateLimiter.getDelay(app))
+
+    case DecreaseDelay(app) => // ignore for now
+
+    case ResetDelay(app) =>
+      rateLimiter.resetDelay(app)
+      updateReceiver ! DelayUpdate(app, rateLimiter.getDelay(app))
+      sender() ! ResetDelayResponse
+  }
+
+  private[this] def receiveTaskStatusUpdate: Receive = {
+    case TaskStatusUpdate(_, taskId, status) =>
+      status match {
+        case MarathonTaskStatus.Terminal(terminal) if !terminal.killed =>
+          sendToSelfForApp(taskId, AddDelay(_))
```

Comment:
@aquamatthias could you add that aesthetic point to the style guide if you and @kolloch agree?  I prefer the placeholder syntax since it makes it clearer that `AddDelay` is a function object and not a case object.